cxx/emissions/get_emission.cc

```diff
@@ -4,21 +4,43 @@
 #include "emissions/get_emission.hh"
 
 #include <cassert>
+#include <random>
+#include <string>
 
 #include "emissions/bitflip.hh"
 #include "emissions/gaussian.hh"
 #include "emissions/simple_string.hh"
 
-EmissionVariant get_emission(const std::string& emission_name) {
-  if (emission_name == "gaussian") {
-    return new GaussianEmission();
-  } else if (emission_name == "simple_string") {
-    return new SimpleStringEmission();
-  } else if (emission_name == "sometimes_gaussian") {
-    return new SometimesGaussian();
-  } else if (emission_name == "sometimes_bitflip") {
-    return new SometimesBitFlip();
+EmissionSpec::EmissionSpec(const std::string& emission_str) {
+  if (emission_str == "gaussian") {
+    emission = EmissionEnum::gaussian;
+    observation_type = ObservationEnum::double_type;
+  } else if (emission_str == "simple_string") {
+    emission = EmissionEnum::simple_string;
+    observation_type = ObservationEnum::string_type;
+  } else if (emission_str == "sometimes_gaussian") {
+    emission = EmissionEnum::sometimes_gaussian;
+    observation_type = ObservationEnum::double_type;
+  } else if (emission_str == "sometimes_bitflip") {
+    emission = EmissionEnum::sometimes_bitflip;
+    observation_type = ObservationEnum::bool_type;
+  } else {
+    assert(false && "Unsupported emission name.");
+  }
+}
+
+EmissionVariant cluster_prior_from_spec(const EmissionSpec& spec,
```

Comment:
get_emission remains a much better name for this function, imho.  If we want consistency, it would be better to rename the DistributionVariant-returning cluster_prior_from_spec to get_distribution (and move it into a file in cxx/distributions/get_distribution).

Reply:
It's nice to have an overloaded function that returns either a distribution or an emission (typed as a distribution) in `CleanRelation::make_new_distribution`. How about renaming this to `get_prior` in both cases? It seems self-documenting enough to me that one is for emissions and the other is for distributions, given how the functions are typed, that we don't need to name them `get_distribution` or `get_emission` (which would then require `CleanRelation::make_new_distribution` to dispatch one or the other with an if/else).

Reply:
Okay, same comment as above -- if you want to do this, please add a bunch of comments explaining the situation.

Reply:
Done.